lib/bundler/vendor/thor/lib/thor/line_editor/readline.rb

```diff
@@ -1,5 +1,5 @@
 begin
-  require "readline"
+  require "readline" unless Gem.win_platform?
```

Comment:
This is vendored code, so this change should be proposed to [the thor project](https://github.com/erikhuda/thor) first. And that will probably involve adding Windows CI in there as well, and thus keep going down the rabbit hole... :sweat_smile:.

Maybe the problem could also be worked around by lazily loading `readline` in thor only when the functionality that needs it is loaded. That way you could probably fix this problem (bundler crashing because of a library not used by bundler being loaded) without having to add Windows support in `thor`.

Thanks for what your doing, by the way!

Reply:
Could we maybe accept/merge this change here anyway, while I start working on getting this understood and fixed in `thor` directly? I wanted to start sending PRs for the first ~10 issues I fixed on bundler on Windows, but need working Windows CI to be able to track the progress :/

Reply:
(The thor tests almost pass on Windows, so I will definitely look into what really is happening with `readline` here in bundler's usage of thor [and how to fix the rest of those tests I'm at it... 🤷‍♀️ ])

Reply:
I will continue my investigation into this at https://github.com/bundler/bundler/issues/6902 👷‍♀️

Reply:
I replaced this change in the vendored code by a folder creation in the CI configuration. See more about the reasons for this in #6902. (I could use the help over there)